workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/InputSpecValidator.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import java.io.File;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.exception.ControllerErrorCodes;
+import io.hyscale.controller.model.HyscaleCommandSpec;
+import io.hyscale.controller.util.ServiceProfileUtil;
+
+@Component
+public class InputSpecValidator implements Validator<HyscaleCommandSpec> {
+    
+    private static final Logger logger = LoggerFactory.getLogger(InputSpecValidator.class);
+
+    @Autowired
+    private ServiceSpecFileValidator serviceSpecFileValidator;
+
+    @Autowired
+    private ServiceSpecSchemaValidator serviceSpecSchemaValidator;
+
+    @Autowired
+    private ProfileFileValidator profileFileValidator;
+
+    @Autowired
+    private ProfileSpecSchemaValidator profileSpecSchemaValidator;
+
```

Comment:
@PostConstruct
public void init(){
 List<Filevalidator> filevalidators= new ArrayList;
fileValidators.add(each);
}

Reply:
Will combine file and spec validators for service spec and profile as mentioned in the one of the comments.